test/cpp/run_tests.sh

```diff
@@ -9,6 +9,7 @@ BUILD_ONLY=0
 RMBUILD=1
 LOGFILE=/tmp/pytorch_cpp_test.log
 XLA_EXPERIMENTAL="nonzero:masked_select"
+XLA_SYNC_BEFORE_ITEM_CALL=1
```

Comment:
I think you also need to `export XLA_SYNC_BEFORE_ITEM_CALL`. You can check `XLA_EXPERIMENTAL ` in this file. 

Might be easier if you just do 
```
export XLA_SYNC_BEFORE_ITEM_CALL=1
```